error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/DirectReturn.java

```diff
@@ -113,13 +122,13 @@ private static Optional<ExpressionTree> tryMatchAssignment(Symbol targetSymbol,
   }
 
   /**
-   * Tells whether inlining the given expression to the associated return statement can be done
-   * safely.
+   * Tells whether inlining the given expression to the associated return statement can likely be
+   * done without changing the expression's return type.
    *
-   * <p>Inlining is generally safe, but in rare cases the operation may have a functional impact.
-   * The sole case considered here is the inlining of a Mockito mock or spy construction without an
-   * explicit type. In such a case the type created depends on context, such as the method's return
-   * type.
+   * <p>Inlining an expression generally does not change its return type, but in rare cases the
+   * operation may have a functional impact. The sole case considered here is the inlining of a
+   * Mockito mock or spy construction without an explicit type. In such a case the type created
+   * depends on context, such as the method's return type.
    */
   private static boolean canInlineToReturnStatement(
```

Comment:
Should now find a better name for this method. Suggestions welcome.

Reply:
Hmm, my concern here is more that `canInline` is actually too strong, since now this method only tests are required, but not sufficient set of conditions to be met. I guess we can just  merge as-is.